src/packet/mod.rs

```diff
@@ -10,6 +10,22 @@
 
 #![macro_use]
 
+/// Used to help compute checksum value of packet
+pub trait PseudoHeader {
+    /// Converts IPv4 or IPv6 specific fields to a checksum value
+    // Should we add a "next_header" u16 argument for ipv6 packets with extension header?
```

Comment:
I think it would be a shame to do anything protocol specific - ideally this abstraction should be protocol agnostic.

If it is 100% necessary, I wouldn't be opposed to IP specific functionality, given their ubiquity.